### PR TITLE
allow locale selection for tree certificate and code

### PR DESCRIPTION
**Pull request by sunilsabatp**
feature
-Provided a dropdown menu to select the locale for issuing the tree certificate and code

## Summary by CodeRabbit

- **New Features**
    - Enhanced user control by adding the ability to select notification language during bulk code issuance.
    - Improved user experience with updated UI for selecting notification language based on the bulk method.

### CODE REVIEW

**Review by mariahosfeld — COMMENTED** [review]
Functionality looks good to me.

File: src/features/user/BulkCodes/forms/IssueCodesForm.tsx [hunk 1]
```diff
@@ -242,6 +254,23 @@ const IssueCodesForm = (): ReactElement | null => {
...
+                    id="demo-simple-select"
+                    value={notificationLang}
+                    label="Notification language"
+                    onChange={handleLocale}
+                  >
+                    <MenuItem value={'en'}>English</MenuItem>
+                    <MenuItem value={'de'}>Deutsch</MenuItem>
+                  </Select>
+                </FormControl>
+              )}
```
Comment (mohitb35) [inline src/features/user/BulkCodes/forms/IssueCodesForm.tsx:273]: Please remove the "demo" id's and adapt the names as necessary.

File: src/features/user/BulkCodes/forms/IssueCodesForm.tsx [hunk 2]
```diff
@@ -242,6 +254,23 @@ const IssueCodesForm = (): ReactElement | null => {
...
+                    labelId="demo-simple-select-label"
+                    id="demo-simple-select"
+                    value={notificationLang}
+                    label="Notification language"
+                    onChange={handleLocale}
+                  >
+                    <MenuItem value={'en'}>English</MenuItem>
+                    <MenuItem value={'de'}>Deutsch</MenuItem>
+                  </Select>
+                </FormControl>
```
Comment (mohitb35) [inline src/features/user/BulkCodes/forms/IssueCodesForm.tsx:272]: Could you structure this using the `<TextField select>` approach that's followed in other places (e.g. src/features/user/ManageProjects/components/BasicDetails.tsx). Note that it does not need to be within a `Controller` in this case.
```
	<TextField
		label={t('manageProjects:ecosystem')}
		variant="outlined"
		select
		onChange={onChange}
		value={value}
		onBlur={onBlur}
		error={errors.metadata?.ecosystem !== undefined}
		helperText={
			errors.metadata?.ecosystem !== undefined &&
			errors.metadata.ecosystem.message
		}
	>
		{ecosystemTypes.map((ecosystem) => (
			<MenuItem key={ecosystem} value={ecosystem}>
				{t(`manageProjects:ecosystemTypes.${ecosystem}`)}
			</MenuItem>
		))}
	</TextField>
```

File: src/features/user/BulkCodes/forms/IssueCodesForm.tsx [hunk 3]
```diff
@@ -242,6 +254,23 @@ const IssueCodesForm = (): ReactElement | null => {
...
+                  </InputLabel>
+                  <Select
+                    labelId="demo-simple-select-label"
+                    id="demo-simple-select"
+                    value={notificationLang}
+                    label="Notification language"
+                    onChange={handleLocale}
+                  >
+                    <MenuItem value={'en'}>English</MenuItem>
+                    <MenuItem value={'de'}>Deutsch</MenuItem>
```
Comment (mohitb35) [inline src/features/user/BulkCodes/forms/IssueCodesForm.tsx:270]: Text is hardcoded and not translatable - this should be addressed for all visible text, including labels, options, placeholders etc.

File: src/features/user/BulkCodes/forms/IssueCodesForm.tsx [hunk 4]
```diff
@@ -242,6 +254,23 @@ const IssueCodesForm = (): ReactElement | null => {
...
                 label={t('bulkCodes:occasion')}
               />
+              {bulkMethod === 'import' && (
+                <FormControl fullWidth>
+                  <InputLabel id="demo-simple-select-label">
+                    Notification language
+                  </InputLabel>
+                  <Select
+                    labelId="demo-simple-select-label"
+                    id="demo-simple-select"
```
Comment (mohitb35) [inline src/features/user/BulkCodes/forms/IssueCodesForm.tsx:264]: While using sample code, please update the id, label etc. accordingly.

File: src/features/user/BulkCodes/forms/IssueCodesForm.tsx [hunk 5]
```diff
@@ -52,7 +60,11 @@ const IssueCodesForm = (): ReactElement | null => {
   const [isSubmitted, setIsSubmitted] = useState(false);
   const [isEditingRecipient, setIsEditingRecipient] = useState(false);
   const [isAddingRecipient, setIsAddingRecipient] = useState(false);
+  const [notificationLang, setNotificationLang] = React.useState('en');
 
+  const handleLocale = (event: SelectChangeEvent) => {
+    setNotificationLang(event.target.value as string);
+  };
```
Comment (mohitb35) [inline src/features/user/BulkCodes/forms/IssueCodesForm.tsx:67]: 1. The name could be more appropriate. `handleLocale` sounds like I'm doing something with translations for the app. `updateNotificationLocale` may be more appropriate.
2. Do we really need a separate handler function? The only thing this does is call `setNotificationLang` which could be done inline within `onChange`. It isn't removing any duplication either.

**Review by mohitb35 — CHANGES_REQUESTED** [review]
@sunilsabatp Please go through my feedback

**Comment by mariahosfeld** [comment]
@sunilsabatp , the starting locale is English, even though I have set the language in the options to German. If the user set their overall language to DE, then also in this feature this needs to be the prefilled value. For all other cases (at the moment), EN is the prefilled setting.

**Review by mariahosfeld — CHANGES_REQUESTED** [review]
Still English preselected, even though the language in the profile is set to German.
Info: My browser is by standard set to English, maybe this is influencing the behavior?

**Comment by sunilsabatp** [comment]
> Still English preselected, even though the language in the profile is set to German. Info: My browser is by standard set to English, maybe this is influencing the behavior?

I'm not sure; however, it's working fine for me.

**Review by mariahosfeld — COMMENTED** [review]
Functionality works for me now.

**Review by mariahosfeld — APPROVED** [review]

**Review by mohitb35 — APPROVED** [review]

**Pull request merged**